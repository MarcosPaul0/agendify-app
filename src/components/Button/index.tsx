import { Text, TouchableOpacity, View } from 'react-native';
import { Spinner } from '@components/Spinner';
import { IButtonProps } from './interfaces/buttonProps.interface';
import { BUTTON_VARIANTS } from './types/buttonVariants.type';

export function Button({
  text,
  isLoading = false,
  variant = 'filled',
  icon,
  isDanger = false,
  ...rest
}: IButtonProps) {
  const primaryButtonColor = rest.disabled ? 'BLUE_100' : 'BLUE_500';

  const buttonColor = isDanger ? 'ERROR' : primaryButtonColor;

  const BUTTON_VARIANT = {
    [BUTTON_VARIANTS.filled]: `h-11 bg-${buttonColor}`,
    [BUTTON_VARIANTS.outlined]: 'h-10 border-2 bg-transparent',
  };

  const TEXT_VARIANT = {
    [BUTTON_VARIANTS.filled]: 'text-GRAY_50',
    [BUTTON_VARIANTS.outlined]: `text-${buttonColor}`,
  };

  return (
    <TouchableOpacity
      className={`
        w-full justify-center my-3 items-center flex-row
        border-${buttonColor} rounded-xl ${BUTTON_VARIANT[variant]}
      `}
      {...rest}
    >
      {isLoading ? (
        <Spinner />
      ) : (
        <>
          {icon && <View className="mr-2">{icon}</View>}
          <Text
            className={`
            font-bold text-center text-base ${TEXT_VARIANT[variant]}
          `}
          >
            {text}
          </Text>
        </>
      )}
    </TouchableOpacity>
  );
}
