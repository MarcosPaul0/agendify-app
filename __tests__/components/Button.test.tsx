import { render } from '@testing-library/react-native';
import { Button } from '@components/Button';

describe('Button', () => {
  it('Match with snapshot', async () => {
    const tree = render(<Button text="fake-text" title="fake-title" />);

    expect(tree).toMatchSnapshot();
  });
});
