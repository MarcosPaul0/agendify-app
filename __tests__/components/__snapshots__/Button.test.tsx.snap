// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Match with snapshot 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignItems": "center",
      "borderBottomLeftRadius": 12,
      "borderBottomRightRadius": 12,
      "borderTopLeftRadius": 12,
      "borderTopRightRadius": 12,
      "flexDirection": "row",
      "height": 44,
      "justifyContent": "center",
      "marginBottom": 12,
      "marginTop": 12,
      "opacity": 1,
      "width": "100%",
    }
  }
  testID="fake-text"
>
  <Text
    style={
      [
        {
          "fontWeight": "700",
        },
        {
          "textAlign": "center",
        },
        {
          "fontSize": 16,
          "lineHeight": 24,
        },
        {
          "color": "#F8F9FA",
        },
      ]
    }
  >
    fake-text
  </Text>
</View>
`;
